include/splinterdb/splinterdb.h

```diff
@@ -292,4 +292,7 @@ splinterdb_iterator_get_current(splinterdb_iterator *iter, // IN
 int
 splinterdb_iterator_status(const splinterdb_iterator *iter);
 
+bool
+splinterdb_validate_key_in_range(const splinterdb *kvs, slice key);
```

Comment:
Add this as extern interface. I used this to write a small unit-test just to sanitize my own code and verify that this validation, indeed, works.

Now, this interface will also be available publicly to end-users who may wish to validate their keys before inserts (given that Splinter is not doing it in optimized builds).

Reply:
This should not be public.

Making this public is just setting a trap for users when we get rid of the min/max key stuff.  When that happens, this API won't even make sense any more.

Reply:
You can test it w/o putting it in `include`, which is the real public API.

Reply:
Reworked as suggested. Renamed routine to simply `validate_key_in_range()` and removed interface from public .h file. Instead, did a slightly sloppy thing of listing the prototype in the single test-case file that needs it. (This works, and good enough for now.)

Reply:
How about creating `src/splinterdb_private.h` and put it in there?

Then have `src/splinterdb_private.h` include `include/splinterdb/splinterdb.h` and have `src/splinterdb.c` include `src/splinterdb_private.h`, similar to how things are done with the btree code.

The test would also include `src/splinterdb_private.h`

Reply:
Hmmm ... @rtjohnso , I know there is a precedence here with `btree.h` but in this case, the `private.h` will initially have just this one extern proto.

I feel it's an over-kill to engineer it this way just to satisfy the one (and probably ever only) consumer of this extern prototype which is the test case. 

On a quick look, I found that somewhere ~10 files are currently #including `splinterdb.h` .  All those would need to be changed to now #include `splinterdb_private.h` -- which will [needlessly] bloat this change-set.

I am not entirely convinced that this is the preferred way, especially given that all this is for just one extern prototype visibility.